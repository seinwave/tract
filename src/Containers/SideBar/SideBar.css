
.sidebar{
    position: fixed; 
}

#sidebar-wrapper {
    margin-top: 1.5rem;
    min-height: 100vh !important;
    width: 100vw;
    margin-left: 1rem;
    -webkit-transition: margin .25s ease-out;
    -moz-transition: margin .25s ease-out;
    -o-transition: margin .25s ease-out;
    transition: margin .25s ease-out;
    border-right: 1px solid #e4e9f0;
    -webkit-box-flex: 0;
    -ms-flex: 0 0 8.333333%;
    flex: 0 0 10%;
    max-width: 10%;
}

#page-content-wrapper {
    min-width:0;
    width:80%
}

.navlink-span{
    display: block;
}