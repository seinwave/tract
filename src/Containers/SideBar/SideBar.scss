// .sidebar {
//     position: fixed;
//     font-size: 20px;
//     background: #fff;
// }



// #sidebar-wrapper {
//     min-height: 100vh !important;
//     padding-top: 25px;
//     background: #fff;
//     width: 100vw;
//     -webkit-transition: margin .25s ease-out;
//     -moz-transition: margin .25s ease-out;
//     -o-transition: margin .25s ease-out;
//     transition: margin .25s ease-out;
//     border-right: 1px solid #215297;
//     -webkit-box-flex: 0;
//     -ms-flex: 0 0 8.333333%;
//     flex: 0 0 10%;
//     max-width: 15%;
// }

// #page-content-wrapper {
//     min-width: 0;
//     width: 80%
// }

// .navlink-span {
//     display: block;
// }