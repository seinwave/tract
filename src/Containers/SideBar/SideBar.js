import React from "react";
import "./SideBar.css";
import Nav from "react-bootstrap/Nav";
import { v4 as uuidv4 } from "uuid";

function SideBar({ children }) {
  return (
    <Nav defaultActiveKey="/home" className="sidebar flex-column">
      {children.map((c) => (
        <div key={uuidv4()} className="navlink-span">
          {c}
        </div>
      ))}
    </Nav>
  );
}

export default SideBar;
