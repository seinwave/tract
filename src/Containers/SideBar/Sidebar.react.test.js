import React from "react";
import SideBar from "./SideBar";
import renderer from "react-test-renderer";

it("renders as expected", () => {
  const tree = renderer
    .create(
      <SideBar>
        <div className="testNavChild">Item</div>
        <div className="testNavChild2">Item2</div>
      </SideBar>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
