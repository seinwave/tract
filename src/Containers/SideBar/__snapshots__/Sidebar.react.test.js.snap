// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="sidebar"
>
  <img
    alt="tract-logo"
    className="topbar-logo-image"
    src="https://1000logos.net/wp-content/uploads/2017/11/Chipotle-Logo.png"
  />
  <a
    className="navbar-brand"
    href="www.mattseidholz.com/"
  >
    Tract
  </a>
  <div
    className="flex-column nav"
    onKeyDown={[Function]}
  >
    <div
      className="navlink-span"
    >
      <div
        className="testNavChild"
      >
        Item
      </div>
    </div>
    <div
      className="navlink-span"
    >
      <div
        className="testNavChild2"
      >
        Item2
      </div>
    </div>
  </div>
</div>
`;
